src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js

```diff
@@ -203,15 +236,52 @@ const QuickMarcEditorRows = ({
 
                 {
                   isContentField && (
-                    <Field
-                      dirty={false}
-                      aria-label={intl.formatMessage({ id: 'ui-quick-marc.record.subfield' })}
-                      name={`${name}[${idx}].content`}
-                      component={ContentField}
-                      marginBottom0
-                      disabled={isDisabled}
-                      id={`content-field-${idx}`}
-                    />
+                    <>
+                      <Field
```

Comment:
I think we can separate a lot of logic related to location lookup into a separate component `<LocationField>`
Similar to `isMaterialCharsField`, `isPhysDescriptionField` or other specific fields let's create a condition `isLocationField` that's going to check if current field is `852` and MARC type is `holdings`.  If that's true - then render `<LocationField>`. Otherwise it's a regular `<ContentField>`

And then in `<LocationField>` we can put all the logic - from storing modal visibility to setting correct field value.

Reply:
Great idea, don't know why I didn't think of it myself. Thanks!